ocp_resources/user_defined_network.py

```diff
@@ -128,3 +139,88 @@ def to_dict(self) -> None:
 
             if self.ipam_lifecycle:
                 _layer2["ipamLifecycle"] = self.ipam_lifecycle
+
+
+class Layer3Subnets:
+    """
+    UserDefinedNetwork layer3 subnets object.
+
+    API reference:
+    https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/#layer3subnet
+    """
+
+    def __init__(
+        self,
+        cidr: Optional[str] = None,
```

Comment:
How come none of the args are mandatory?

Reply:
Because the base API itself 'subnets' is not mandatory.

Reply:
OK, I was wrong,  'subnets' is indeed mandatory, my confusion for that I relied on the API documentation which didn't state it is mandatory, but I know it is mandatory because when I tried to create UDN resource whithout it I got:
```
 The UserDefinedNetwork "new-net" is invalid: spec.layer3: Invalid value: "object": Subnets is required for Layer3 topology
```
I will fix it.

Reply:
Done.